### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by AliRezaBmeDu**
In this branch, we have done the following changes.

- Create module to store the people, books and rentals data when a session ends
- Create methods for loading data from the existing json files
- Preserve the data in json format
- Address the conditions of non-existance of json files

### CODE REVIEW

File: explain.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+date = Date.now
+puts date
+
+# puts date.strftime('%y/%m/%d')
+
+# puts date.strftime('%H/%M/%S')
```
Comment (Ibizugbe) [inline explain.rb]: Nice job so far😉. As we have learned in the [ruby best practices requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md), Kindly take away every commented piece of code as this makes your codebase look messy and gives it a less professional look😉.

image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @Team👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] When the app is exited, the data is saved to the json file👍🏼.
- [x] When the app is start up, the data is retrieved from the json file👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: explain.rb (hunk 1 above)
Comment (AliRezaBmeDu) [reply explain.rb]: @Ibizugbe , The file was to test out some ruby commands. It does not participate in the project. Anyway, I am deleting it.

**Review by zeff96 — APPROVED** [review]
### Status: APPROVED :green_circle: 
Hi @AliRezaBmeDu and @Mike47ip , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
congrats

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: Correct GitFlow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**